api/v1/helpers.go

```diff
@@ -1392,6 +1392,16 @@ func (v *VerticaDB) GetVerticaUser() string {
 	return vmeta.GetSuperuserName(v.Annotations)
 }
 
+// GetPasswordSecret returns the password secret
+func (v *VerticaDB) GetPasswordSecret() string {
+	// status holds the current password
+	if v.Status.PasswordSecret != "" {
```

Comment:
Issue with this: if I first create a db with empty password (password secret empty), then I set the passwordSecret, I will be getting the secret from spec instead of status and that will be a problem until password change is done in the db.
I think if v.IsDBInitialized() is false, we should get the secret from spec and then when db is initialized, we get it from spec.

We also need to carefully think about backward compatibility. If the user had a passwordsecret in an older version and they upgrade, status will be empty, and we will get the secret from status which will be wrong. 
Let's discuss about how to handle this case.

Reply:
As discussed, changed status.passwordSecret to be a string pointer. We only get password from status if it's not nil. This will cover the db creation, db revive, and operator upgrade cases.